examples/tutorials/self-paced-training/part-3_security_and_privacy/chapter-5_Privacy_In_Federated_Learning/05.1_privacy_filter/privacy_filtering.ipynb

```diff
@@ -1,19 +1,102 @@
+    "\n",
+    "The behavior can be selected by using the [FilterType](https://nvflare.readthedocs.io/en/main/apidocs/nvflare.job_config.defs.html#nvflare.job_config.defs.FilterType). Users must specify the filter type as either `FilterType.TASK_RESULT` (flow from executor to controller) or `FilterType.TASK_DATA` (flow from controller to executor).\n",
+    "\n",
+    "The filter will be added \"task_data_filters\" and task_result_filters accordingly and be applied to the specified tasks (defaults to “[*]” for all tasks).\n",
+    "\n",
+    "For example, you can add a privacy filter as such.\n",
+    "```python\n",
+    "pp_filter = PercentilePrivacy(percentile=10, gamma=0.01)\n",
+    "job.to(pp_filter, \"site-1\", tasks=[\"train\"], filter_type=FilterType.TASK_RESULT)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11e4f07a-2f0c-4de6-83f6-15c13f693b0a",
+   "metadata": {},
+   "source": [
+    "#### Enforcing of Filters\n",
+    "Data owners can enforce filters to be applied to any job they execute. Enforcing filters on all executed jobs ensures that data owners maintain control over privacy and compliance. This can be useful for several reasons:  \n",
+    "\n",
+    "- **Consistent Privacy Protection:** Ensures that every model update follows predefined privacy policies, reducing the risk of accidental data leakage.  \n",
+    "- **Regulatory Compliance:** Helps meet legal and ethical standards (e.g., HIPAA, GDPR) by enforcing data anonymization or masking sensitive information.  \n",
+    "- **Defense Against Emerging Threats:** Provides a safeguard against evolving attack techniques, such as model inversion, membership inference, or detection of malicious model weights.  \n",
+    "- **Customization for Sensitive Data:** Allows data owners to tailor privacy mechanisms to their specific data types, ensuring that only necessary information is shared.  \n",
+    "- **Trust and Collaboration:** Encourages participation in Federated Learning by reassuring institutions that their data remains secure throughout the process.  \n",
+    "\n",
+    "By enforcing privacy filters in NVFlare, data owners can ensure a reliable and secure FL environment without relying solely on external safeguards. For more details, see the [documentation](https://nvflare.readthedocs.io/en/main/user_guide/security/site_policy_management.html#privacy-management)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "351e067f-495a-45d9-bcfd-d8031584cffb",
+   "metadata": {},
+   "source": [
+    "#### Writing Your Own Filter\n",
+    "For writing your own filter, you can utilize the [DXOFilter](https://nvflare.readthedocs.io/en/main/apidocs/nvflare.apis.dxo_filter.html#nvflare.apis.dxo_filter.DXOFilter) base class. For details see the [documentation](https://nvflare.readthedocs.io/en/main/programming_guide/filters.html). "
+   ]
```

Comment:
might be good to add a link to the next notebook after each notebook ends

Reply:
done